src/Avalonia.Base/Platform/Storage/FileIO/StorageBookmarkHelper.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Buffers;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+
+namespace Avalonia.Platform.Storage.FileIO;
+
+/// <summary>
+/// In order to have unique bookmarks across platforms, we prepend a platform specific suffix before native bookmark.
+/// And always encoding them in base64 before returning to the user.
+/// </summary>
+/// <remarks>
+/// Bookmarks are encoded as:
+/// 0-15  - platform key
+/// 16+ - native bookmark value
+/// Which is then encoded in Base64.
+/// </remarks>
+internal static class StorageBookmarkHelper
+{
+    private const int PrefixLength = 16;
+
+    private static ReadOnlySpan<byte> FakeBclBookmarkPlatform => "avalonia-bcl"u8;
```

Comment:
nit: won't it make more sense to have this prefix in the BCL provider?

Reply:
I intended to make EncodeBclBookmark/TryDecodeBclBookmark methods visible outside of BclStorageProvider. Like it's used in macOS backend now:
https://github.com/AvaloniaUI/Avalonia/pull/16090/files/47c3e3cdfa3496a46240fd5141358c81b7b5f340#diff-01a97878717fd9596b20885a6a096cda57ec8f4b1c98d00961d5a1fa348d2c77R137-R140